lib/manager/gradle/build-gradle.ts

```diff
@@ -212,7 +212,12 @@ export function collectVersionVariables(
     }
 
     if (!dep.currentValue && variables[depName]) {
-      dep.currentValue = variables[depName];
+      const isVersionLiteral = !/^[a-zA-Z_][a-zA-Z0-9_.]*$/.test(
```

Comment:
This regex looks strange to me, what does it test?

Reply:
Sometimes packages have strange versioning, e.g. this one:
`org.springframework.cloud:spring-cloud-dependencies:Hoxton.SR8`

Will your solution handle that?

Reply:
To what extend can we leverage our existing gradle versioning to determine if a version is valid?

Reply:
Variable name can be valid version too. I'm planning to change #6477 implementation a little bit, such that we don't need to care about this.

Reply:
@zharinov should that change be done first?

Reply:
Do you mean separate PR? It will break existing extractions, so pretty much PRs will be closed